relentless/simulate/hoomd.py

```diff
@@ -801,6 +801,53 @@ def reset(self):
             # rdf will be reset on next call
 
 
+class WriteTrajectory(simulate.AnalysisOperation):
+    """Writes a .gsd file.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the .gsd file to be written.
+    every : int
+        Interval of time steps at which to write a snapshot of the simulation.
+    velocity : bool
+        Log particle velocity.
+    image : bool
+        Log particle image.
+    typeid : bool
+        Log particle type ID.
+    mass : bool
+        Log particle mass.
+
+    """
+
+    def __init__(self, filename, every, velocity, image, typeid, mass):
+        self.filename = filename
+        self.every = every
+        self.velocity = velocity
+        self.image = image
+        self.typeid = typeid
+        self.mass = mass
+
+    def __call__(self, sim):
+        _dynamic = ["property"]
+        if self.velocity is True or self.image is True:
+            _dynamic.append("momentum")
+
+        with sim.hoomd:
+            all_ = hoomd.group.all()
+            hoomd.dump.gsd(
+                self.filename,
+                period=self.every,
+                group=all_,
+                overwrite=True,
+                dynamic=_dynamic,
+            )
+
+    def finalize(self, sim):
+        return super().finalize(sim)
```

Comment:
No need to reimplement
```suggestion
```

Reply:
If this not implemented I get an error. 

`Traceback (most recent call last):
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/tests/simulate/test_hoomd.py", line 267, in test_writetrajectory
    sim = h.run(pot, self.directory)
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/relentless/simulate/simulate.py", line 190, in run
    op(sim)
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/relentless/simulate/simulate.py", line 71, in __call__
    return self._op(sim)
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/relentless/simulate/hoomd.py", line 413, in __call__
    analyzer(sim)
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/relentless/simulate/simulate.py", line 70, in __call__
    self._ensure_op(sim)
  File "/Users/levipetix/Documents/research/projects/relentless_inverese_design/relentless/relentless/simulate/simulate.py", line 86, in _ensure_op
    self._op = BackendOp(*self._args, **self._kwargs)
TypeError: Can't instantiate abstract class WriteTrajectory with abstract method finalize`

Reply:
Oh I see. This is because `finalize` is abstract in the parent class, so it needs to be implemented. However, the body should just be `pass` to mean "do nothing" rather than `super()`. I'm actually not sure what the latter behavior even is.